Sources/SwiftParser/Availability.swift

```diff
@@ -253,6 +253,23 @@ extension Parser {
     return self.consumeAnyToken()
   }
 
+  /// Consume the unexpected version token(e.g. integerLiteral, floatingLiteral, identifier) until the period no longer appears.
+  private mutating func parseUnexpectedVersionTokens() -> RawUnexpectedNodesSyntax? {
+    var unexpectedTokens: [RawSyntax] = []
+    var keepGoing: RawTokenSyntax? = nil
+    var loopProgress = LoopProgressCondition()
+    repeat {
+      if let keepGoing {
+        unexpectedTokens.append(RawSyntax(keepGoing))
+      }
+      if let unexpectedVersion = self.consume(if: .integerLiteral, .floatingLiteral, .identifier) {
+        unexpectedTokens.append(RawSyntax(unexpectedVersion))
+      }
+      keepGoing = self.consume(if: .period)
+    } while keepGoing != nil && loopProgress.evaluate(currentToken)
```

Comment:
You could just make `unexpectedTokens` a `[RawTokenSyntax]`, which simplifies the code a little

```suggestion
    var unexpectedTokens: [RawTokenSyntax] = []
    var keepGoing: RawTokenSyntax? = nil
    var loopProgress = LoopProgressCondition()
    repeat {
      if let keepGoing {
        unexpectedTokens.append(keepGoing)
      }
      if let unexpectedVersion = self.consume(if: .integerLiteral, .floatingLiteral, .identifier) {
        unexpectedTokens.append(unexpectedVersion)
      }
      keepGoing = self.consume(if: .period)
    } while keepGoing != nil && loopProgress.evaluate(currentToken)
```

Reply:
I have committed your suggestion, thank you!